src/rgw/rgw_common.h

```diff
@@ -898,6 +898,7 @@ struct RGWBucketInfo {
   bool has_instance_obj{false};
   RGWObjVersionTracker objv_tracker; /* we don't need to serialize this, for runtime tracking */
   RGWQuotaInfo quota;
+  bool deleted{false};
```

Comment:
we had discussed moving this `deleted` flag to an xattr to avoid changing the encoding, but it looks like it could be a flag in `RGWBucketFlags` for `RGWBucketInfo::flags`. that wouldn't change the encoding either

Reply:
done!